google/cloud/storage/oauth2/authorized_user_credentials.h

```diff
@@ -92,16 +115,23 @@ class AuthorizedUserCredentials : public Credentials {
 
     // TODO(#516) - use retry policies to refresh the credentials.
     auto response = request_.MakeRequest(payload_);
-    if (200 != response.status_code) {
+    if (response.status_code >= 300) {
+      return false;
+    }
+    nl::json access_token = nl::json::parse(response.payload, nullptr, false);
+    if (access_token.is_discarded() or access_token.count("token_type") == 0U or
```

Comment:
From the docs, `is_discarded()` only returns true during parsing with a custom callback.  I think `is_null()` is the appropriate check here, no?

Reply:
I tested (and committed a test), and `is_discarded()` is the right thing.

Reply:
Ah, indeed -- that seems to go directly against what the docs say, but the docs are just stale.  I left a comment on an issue in the repo [here](https://github.com/nlohmann/json/issues/458#issuecomment-426868730).